### PR TITLE
Updating results for Firefox 18 to Firefox 23

### PR DESCRIPTION
**Pull request by afmenez**
Updating results for Firefox 18 to Firefox 23.
A few corrections, no surprises.

### CODE REVIEW

File: data-non-standard.js [hunk 1]
```diff
@@ -1218,7 +1212,7 @@ exports.tests = [
     firefox12: false,
     firefox16: true,
     firefox17: false,
-    firefox20: true,
+    firefox20: false,
     firefox28: false,
```
Comment (chicoxyzzy) [inline data-non-standard.js:1257]: this is useless because your PR removes Firefox 28. Don't sure if it's good idea though

Comment (afmenez) [reply data-non-standard.js:1257]: I am not sure what you mean.  This is fixing the result for Firefox 20..27, which was wrong.
No browsers were removed, but I did consolidate identical columns of the table.

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: Firefox 28 was removed by you from browser list for `non-standard` tab. Look at diff starting at line 79. There is only `firefox20` (which includes version from 20 to 29) and `firefox28` anymore in your branch

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: so this particular line (`firefox28: false,`) does nothing

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: in fact there's a lot of `firefox28` mentions in this file

Comment (afmenez) [inline data-non-standard.js:1257]: That's what I said, I removed the column but not the data.  Do you mean you want the columns back?  I am starting to feel optimizing identical columns is a lost battle.

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: What I'm trying to say is that in your branch

```
firefox20: false,
firefox28: false,
```

is the same as

```
firefox20: false,
```

because there is no data under key `firefox28` in browsers list, it is represented under `firefox20` in your branch. So `firefox28: false,` is useless and should be deleted as well as other `firefox28` mentions in this file.

By the way I  hope this PR will not be merged. Those browsers are  **very** obsolete and they don't get rendered in table anyway. The data about particular browsers may still be useful, for example for cases described in https://github.com/kangax/compat-table/issues/785

Comment (afmenez) [reply data-non-standard.js:1257]: I see now what you mean.  I tried to remove redundant data in a previous patch, and had to revert it because I was throwing data away.  Apparently having these 2 redundant lines mean that both browsers were actually tested.  There's a lot of this in this file.  I kept both because I didn't wat to have my patch refused again because of the optimization.

These browsers ARE rendered in this table if you check the "Obsolete Browsers" checkbox.  For the other tables, the data is there and is wrong, so I believe it is worth fixing even if they are not displayed by the default renderer.

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: > Apparently having these 2 redundant lines mean that both browsers were actually tested.
> 
> I kept both because I didn't wat to have my patch refused again because of the optimization.

All browsers which have identical `full` field will nest data from previous marked version's value for each test and it's normal in this repo. Actually you loose data when you delete browser from browser list and not loose any useful data when you delete repeating values in tables (and especially you loose nothing when you delete data for browser not represented in browser list).

> These browsers ARE rendered in this table if you check the "Obsolete Browsers" checkbox. For the other tables, the data is there and is wrong, so I believe it is worth fixing even if they are not displayed by the default renderer.

There are no Firefox browsers with versions lower then 31 get rendered if you check obsolete browsers checkbox on. This is because of `obsolete: very,` fields in very old versions. There is nothing about very obsolete browsers in DOM.
So this is not really an optimisation.

Comment (afmenez) [reply data-non-standard.js:1257]: > All browsers which have identical full field will nest data from previous marked version's value for each test and it's normal in this repo. Actually you loose data when you delete browser from browser list and not loose any useful data when you delete repeating values in tables (and especially you loose nothing when you delete data for browser not represented in browser list).

I agree with you, and I said it above.  I'll do a separate patch removing redundant entries like this, please help me merge it then.

> There are no Firefox browsers with versions lower then 31 get rendered if you check obsolete browsers checkbox on. This is because of obsolete: very, fields in very old versions. There is nothing about very obsolete browsers in DOM.  So this is not really an optimisation.

I understand how this works, and I have marked some versions as "very obselete" myself.  I even have an open issue suggesting we mark more versions as "very obsolete".  But for non-standard features table, which is the one we are talking about here (data-non-standard.js), no browsers are marked as "very obsolete".

Comment (chicoxyzzy) [reply data-non-standard.js:1257]: > But for non-standard features table, which is the one we are talking about here (data-non-standard.js), no browsers are marked as "very obsolete".

Wow. Didn't notice. They should match obsolete browsers for other tables IMHO. Did someone had any counterarguments?

Comment (afmenez) [inline data-non-standard.js:1257]: > Wow. Didn't notice. They should match obsolete browsers for other tables IMHO. Did someone had any counterarguments?

What we should have is a single browser definition place, like in
"esnext-browsers.js", and remove all other form inside the data files.  I
an willing to do that in a separate patch, but only after this pull queue
is emptied to avoid all the merge issues.

**Comment by afmenez** [comment]
So, this is another straightforward patch and all opposition was addressed. What is missing to have it merged?

**Comment by chicoxyzzy** [comment]
This needs rebase. It LGTM but I'd prefer someone else to review it (I think @ljharb may disagree with merging FF versions in `non-standard` tab for example)

**Comment by afmenez** [comment]
I don't have access to my main machine now, I'll rebase it tomorrow.  If the column merge is the only thing holding this PR back, I'll revert it when I do the update.
I'll also do another PR regarding FF49, which was released today.

File: data-non-standard.js [hunk 2]
```diff
@@ -87,14 +87,8 @@ exports.browsers = {
...
   firefox20: {
-    full: 'Firefox 20-27',
-    short: 'FF 20-27',
-    family: 'SpiderMonkey',
-    obsolete: true
-  },
-  firefox28: {
-    full: 'Firefox 28-29',
-    short: 'FF 28-29',
+    full: 'Firefox 20-29',
```
Comment (ljharb) [inline data-non-standard.js:90]: It would be nice to keep all of the explicit `firefox28` results, but i don't think it needs to block the PR.

**Comment by chicoxyzzy** [comment]
I'd prefer to merge this before merging all browsers descriptions into one module (as per discussed in https://github.com/kangax/compat-table/issues/861)

File: data-non-standard.js (hunk 2 above)
Comment (afmenez) [reply data-non-standard.js:90]: All the data is intact, this only merges the visualization.

**Comment by afmenez** [comment]
Rebased and ready to merge.

**Comment by chicoxyzzy** [comment]
Thank you
